library_system.cpp

```diff
@@ -106,7 +106,7 @@ class  user_data {
     void return_book(book_details book)
     {
         book_history[book] = "Available";
-        status[book.id].second = 0;
+        status[book.id].second = 0;          // here whhy are you book.id earlier you have book as the key of map ?

```

Comment:
sorry that was my mistake I had updated book by book id I had committed the wrong code